src/lib/MakeEvalable.ts

```diff
@@ -0,0 +1,167 @@
+      else if (plugin.factory) yield plugin.factory.name;
+      else yield `new ${constructor.name}`;
+      yield new AnnotatedToken("(", "open");
+      yield* makeListTokens(plugin.destructureFunc(obj));
+      yield new AnnotatedToken(")", "close");
+    }
+  } else {
+    // any other primitive type
+    yield JSON.stringify(obj);
+  }
+}
+
+function* iterateWithLookahead<T>(
+  iterable: Iterable<T>,
+  func: (element: T, lookahead: T | undefined) => Generator<T>,
+): Generator<T> {
+  let element: undefined | T = undefined;
+  for (const lookahead of iterable) {
+    if (element !== undefined) yield* func(element, lookahead);
+    element = lookahead;
+  }
+  if (element !== undefined) yield* func(element, undefined);
+}
+
+function* wrapTokens(tokenGen: TokenGenerator): TokenGenerator {
+  let curIndent = "\n";
+  yield* iterateWithLookahead(
+    tokenGen,
+    function* (curToken: Token, nextToken: Token | undefined) {
+      const curTokenType = (curToken as AnnotatedToken).type;
+      const nextTokenType = (nextToken as AnnotatedToken)?.type;
+      if (curTokenType == "open") {
+        yield curToken;
+        if (nextTokenType != "close") {
+          // keep empty structures on a single line
+          curIndent = curIndent + relIndent;
+          yield curIndent;
+        }
+      } else {
+        if (curTokenType != "delimiter" || nextTokenType != "close")
+          // output only non-trailing delimiter
+          yield curToken;
+        if (nextTokenType == "close")
+          curIndent = curIndent.slice(0, -relIndent.length);
+        if (curTokenType == "delimiter") yield curIndent;
+      }
+    },
+  );
+}
+
+export function makeEvalable(obj: unknown, multiline = true) {
+  let tokenGen = makeEvalableTokens(obj);
+  if (multiline) tokenGen = wrapTokens(tokenGen);
+  return [...tokenGen]
+    .map((t) => (t instanceof AnnotatedToken ? t.text : t)) // remove annotations
+    .join("");
+}
+
+/**
+ * Register Custom types that shall be renderable to an evalable string.
+ *
+ * @param constructorFunc This has to be either the constructor (i.e. <Date>)
+ *                        or a factory function (i.e. <dayjs>)
+ * @param destructureFunc This has to be a function that deconstructs an
+ *                        instance of the class into the parameters that were
+ *                        passed to the constructor.
+ * @param options         Sometimes an object shall not be created via
+ *                        constructor but via factory. In this case the factory
+ *                        func can be passed here.
+ *                        Furthermore optionally a different name can be
+ *                        specified.
+ */
+export function registerCustomType<T extends object>(
+  constructorFunc: ConstructorFunc<T>,
+  destructureFunc: DestructureFunc<T>,
+  options?: {
+    factory?: FactoryFunc<T>;
+    name?: string;
+  },
+) {
+  if (customTypePlugins.has(constructorFunc)) return false;
```

Comment:
I did so in my first implementation. But when registering a plugin from within a svelte module it seems it could be called twice (didn't follow up why).

But I will change this back to raising an error...